snuba/datasets/outcomes_processor.py

```diff
@@ -37,6 +29,8 @@ def process_message(
                 datetime.strptime(value["timestamp"], settings.PAYLOAD_DATETIME_FORMAT),
             ),
             "outcome": value["outcome"],
+            "category": value.get("category", DataCategory.ERROR),
```

Comment:
i'm wondering if we should log a warning if there's no category on the event -- we should always be receiving it from places where we emit it now.

Reply:
(the same could be said for the rest of the fields where we expect a value but set it if it doesn't exist)

Reply:
You could always add a metric.

Reply:
Sending metrics is good.
About logging a warning in a consumer, please be very careful about that. Each warn is creating a new event that will have to be ingested. IF we start doing that at high volume on outcomes, we could be overloading the errors consumer. 
So a good practice is never to log warning/errors from a processor/consumer unless you are sure that it could only be logged by a small subset of the messages.

Reply:
metrics sounds like the way to go, will add them. thank you!

Reply:
when downstream would it be converted to a string? it seemed to work okay in the unit test and e2e environment tests when leaving the category field off of the input json. regardless, i can replace it with `DataCategory.ERROR.value` and should work.

Reply:
Actually you may be right, I forgot that is an integer. Let me check

Reply:
```
>>> a = {"a": "b", "c": DataCategory.ERROR}
>>> rapidjson.dumps(a)
'{"a":"b","c":1}'
```

You are right, please disregard my comment.